src/cli/adapter/eslint.ts

```diff
@@ -29,26 +29,51 @@ export interface ESLintOutput {
 export const runESLint = async (
   mode: 'format' | 'lint',
   logger: Logger,
+  overrideConfigFile?: string,
 ): Promise<ESLintOutput> => {
   logger.debug('Initialising ESLint...');
 
+  const cwd = process.cwd();
+
+  const ESLint = await loadESLint({ useFlatConfig: true });
   const engine = new ESLint({
     cache: true,
     fix: mode === 'format',
-    reportUnusedDisableDirectives: 'error',
+    overrideConfigFile,
+    overrideConfig: {
+      linterOptions: {
+        reportUnusedDisableDirectives: true,
+      },
+    },
   });
 
-  const cwd = process.cwd();
-
   logger.debug('Processing files...');
 
   const start = process.hrtime.bigint();
 
-  const [formatter, results] = await Promise.all([
+  const [formatter, { type, results }] = await Promise.all([
     engine.loadFormatter(),
-    engine.lintFiles('.'),
+    engine
+      .lintFiles([])
+      .then((r) => ({ type: 'results', results: r }) as const)
```

Comment:
I dislike the use of .then(), could we avoid if possible? Thoughts? @72636c

Reply:
:+1: I can wrap this in a function